### PR TITLE
Improve coordinate validation

### PR DESCRIPTION
**Pull request by tjstienstra**
#### Brief description of what is fixed or changed
This PR improves the validation of generalized coordinates, by also checking whether a symbol is used in both the generalized coordinates and generalized speeds.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by TJStienstra** [comment]
6ce25629d14ef45748f166d90ca8c9808b6ae65a changed the `Joint.kdes` property to return it's negative, such that `KanesMethod(..., explicit_kinematics=False).mass_matrix_kin` would return a positive identity matrix, which is what would normally be expected. 00f11045b86ed083b293518894c152e0c9a53e5f reverts this change, because `q' - u` is not mathematical equivalent to `u - q'`. Though it should be noted that these expressions should always be equal to zero.

File: sympy/physics/mechanics/functions.py [hunk 1]
```diff
@@ -756,6 +756,10 @@ def _validate_coordinates(coordinates=None, speeds=None, check_duplicates=True,
         if len(speeds) != len(set(speeds)):
             raise ValueError('Duplicate generalized speeds found, all '
                              'generalized speeds should be unique.')
+        if len(coordinates) + len(speeds) != len(
+            set(coordinates).union(speeds)):
+            raise ValueError('Some generalized coordinates are also defined as '
+                             'generalized speeds.')
```
Comment (brocksam) [inline sympy/physics/mechanics/functions.py:762]: Are we happy this this error message is informative enough? Or do we want to try and also tell the reader which is duplicated?

File: sympy/physics/mechanics/functions.py [hunk 2]
```diff
@@ -750,12 +750,22 @@ def _validate_coordinates(coordinates=None, speeds=None, check_duplicates=True,
...
-                             'generalized speeds should be unique.')
+        seen = set()
+        coord_duplicates = {x for x in coordinates if x in seen or seen.add(x)}
+        seen = set()
+        speed_duplicates = {x for x in speeds if x in seen or seen.add(x)}
+        overlap = set(coordinates).intersection(speeds)
+        if coord_duplicates:
+            raise ValueError(f'The generalized coordinates {coord_duplicates} '
+                             f'have duplicates, all generalized coordinates '
+                             f'should be unique.')
```
Comment (brocksam) [inline sympy/physics/mechanics/functions.py:761]:
```suggestion
            raise ValueError(f'The generalized coordinates {coord_duplicates} '
                             f'are duplicated, all generalized coordinates '
                             f'should be unique.')
```

File: sympy/physics/mechanics/functions.py [hunk 3]
```diff
@@ -750,12 +750,22 @@ def _validate_coordinates(coordinates=None, speeds=None, check_duplicates=True,
...
+        if speed_duplicates:
+            raise ValueError(f'The generalized speeds {speed_duplicates} have '
+                             f'duplicates, all generalized speeds should be '
+                             f'unique.')
```
Comment (brocksam) [inline sympy/physics/mechanics/functions.py:765]:
```suggestion
            raise ValueError(f'The generalized speeds {speed_duplicates} are '
                             f'duplicated, all generalized speeds should be '
                             f'unique.')
```

File: sympy/physics/mechanics/functions.py [hunk 4]
```diff
@@ -750,12 +750,22 @@ def _validate_coordinates(coordinates=None, speeds=None, check_duplicates=True,
...
+        if overlap:
+            raise ValueError(f'{overlap} are defined as both generalized '
+                             f'coordinates as well as generalized speeds.')
```
Comment (brocksam) [inline sympy/physics/mechanics/functions.py:768]:
```suggestion
            raise ValueError(f'{overlap} are defined as both generalized '
                             f'coordinates and generalized speeds.')
```

**Review by brocksam — APPROVED** [review]

**Pull request merged**